docs/guide/search/extend_elasticsearch.md

```diff
@@ -291,7 +291,210 @@
                 - { name: ezplatform.search.elasticsearch.query.location.sort_clause_visitor }
     ```
 
+    ## Custom Aggregation
+    
+    To create a custom aggregation for use with Elasticsearch, create an aggregation class.
+    In the following example, an aggregation groups Location query results according to the Location priority:
+    
+    ``` php
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Query\Aggregation;
+
+    use eZ\Publish\API\Repository\Values\Content\Query\Aggregation\AbstractRangeAggregation;
+    use eZ\Publish\API\Repository\Values\Content\Query\Aggregation\LocationAggregation;
+
+    final class PriorityRangeAggregation extends AbstractRangeAggregation implements LocationAggregation
+    {
+
+    }
+    ```
```

Comment:
The most important information is missing here: what is and what does the aggregation class. 

For instance:

> To provide support for custom aggregation, you will need to create aggregation definition by implementing  `\eZ\Publish\API\Repository\Values\Content\Query\Aggregation` interface  or inherit one of following abstract classes:
> 
> *   `\eZ\Publish\API\Repository\Values\Content\Query\Aggregation\AbstractRangeAggregation`
> *   `\eZ\Publish\API\Repository\Values\Content\Query\Aggregation\AbstractStatsAggregation`
> *   `\eZ\Publish\API\Repository\Values\Content\Query\Aggregation\AbstractTermAggregation`
> 
> Aggregation definition needs contain at least aggregation name and optional aggregation parameters e.g. path string used to limit aggregation results to specific subtree or content type identifier / field definition identifier which will be mapped latter on to search index field name. Aggregation definition should be independent of used search engine....

Latter on we should describe propose of 

*   `\eZ\Publish\API\Repository\Values\Content\Query\Aggregation\FieldAggregation`
*   `\eZ\Publish\API\Repository\Values\Content\Query\Aggregation\LocationAggregation`
*   `\eZ\Publish\API\Repository\Values\Content\Query\Aggregation\RawAggregation`

interfaces, explain used naming convention and then we can go with example :wink: